ui/packages/catalog-ui-search/src/main/webapp/component/location-old/keyword.js

```diff
@@ -0,0 +1,79 @@
+const React = require('react');
+const Announcement = require('component/announcement');
+
+const { AutoComplete } = require('./inputs');
+const Polygon = require('./polygon');
+
+const fetch = require('./inputs/fetch');
+
+class Keyword extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            value: '',
+            loading: false
+        };
+        this.fetch = this.props.fetch || fetch;
+    }
+    async onChange({ id, name }) {
+        this.setState({ value: name, loading: true });
+        const res = await this.fetch(`/search/catalog/internal/geofeature?id=${id}`);
```

Comment:
❓ Should we be doing anything with the status code and such here? If something goes wrong with the request, they'll only find out because of an
```
Invalid Feature
Unrecognized feature type: 
```
error box. And `JSON.stringify(type)` will just return nothing

Reply:
Or will it just actually throw an exception because `undefined has no attribute type` when it tries to enter that switch statement there?